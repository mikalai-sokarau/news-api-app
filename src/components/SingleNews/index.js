import React from "react";
import { ThemeProvider } from "styled-components";
import { standart, favorite, StyledStar } from "./style";

const NewSingle = ({
  checked,
  removeNewsFromFavorite,
  addNewsToFavorite,
  item
}) => (
  <div className="col s4 hoverable">
    <div className="card small">
      <div className="card-image" style={{ position: "relative" }}>
        <img src={item.urlToImage} alt={item.title} />
        <span className="card-title">{item.source.name}</span>
        <ThemeProvider theme={checked ? favorite : standart}>
          <StyledStar
            onClick={
              checked
                ? () => removeNewsFromFavorite({ id: item.url })
                : () => addNewsToFavorite({ id: item.url, data: item })
            }
          />
        </ThemeProvider>
      </div>
      <div className="card-content">
        <p>{item.title}</p>
      </div>
      <div className="card-action">
        <a href={item.url} target="_blank">
          FULL ARTICLE
        </a>
      </div>
    </div>
  </div>
);

export default NewSingle;
