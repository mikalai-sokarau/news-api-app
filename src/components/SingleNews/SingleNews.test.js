import React from 'react';
import { shallow, mount } from 'enzyme';
import SingleNews from './index';

describe('<SingleNews />', () => {
  const fakeItem = { urlToImage: './fake', title: 'fake', source: { name: 'fake' } };

  test('should be rendered', () => {
    expect(
      shallow(
        <SingleNews
          checked
          item={fakeItem}
          removeNewsFromFavorite={() => {}}
          addNewsToFavorite={() => {}}
        />
      )
    ).toMatchSnapshot();
  });

  test('should render `removeNewsFromFavorite` if pass `checked = true`', () => {
    const callback = jest.fn();
    const wrapper = mount(
      <SingleNews
        checked
        item={fakeItem}
        removeNewsFromFavorite={callback}
        addNewsToFavorite={() => {}}
      />
    );

    wrapper.find('ThemeProvider').simulate('click');
    expect(callback).toHaveBeenCalledTimes(1);
    wrapper.find('ThemeProvider').simulate('click');
    wrapper.find('ThemeProvider').simulate('click');
    expect(callback).toHaveBeenCalledTimes(3);
  });

  test('should render `addNewsToFavorite` if pass `checked = false`', () => {
    const callback = jest.fn();
    const wrapper = mount(
      <SingleNews
        checked={false}
        item={fakeItem}
        removeNewsFromFavorite={() => {}}
        addNewsToFavorite={callback}
      />
    );

    wrapper.find('ThemeProvider').simulate('click');
    expect(callback).toHaveBeenCalledTimes(1);
    wrapper.find('ThemeProvider').simulate('click');
    wrapper.find('ThemeProvider').simulate('click');
    expect(callback).toHaveBeenCalledTimes(3);
  });
});
