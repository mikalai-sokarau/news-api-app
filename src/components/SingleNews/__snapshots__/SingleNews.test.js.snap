// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleNews /> should be rendered 1`] = `
<div
  className="col s4"
>
  <div
    className="card small hoverable"
  >
    <styled.div
      className="card-image"
    >
      <img
        alt="fake"
        src="./fake"
      />
      <span
        className="card-title"
      >
        fake
      </span>
      <ThemeProvider
        theme={
          Object {
            "color": "gold",
          }
        }
      >
        <Styled(FaStar)
          onClick={[Function]}
        />
      </ThemeProvider>
    </styled.div>
    <div
      className="card-content"
    >
      <p>
        fake
      </p>
    </div>
    <div
      className="card-action"
    >
      <a
        target="_blank"
      >
        FULL ARTICLE
      </a>
    </div>
  </div>
</div>
`;
